tables/process.py

```diff
@@ -0,0 +1,148 @@
+        :return: None
+        """
+        self._strip_whitespace(table)
+        self._lift_header(table)
+        self._remove_merged_headers(table)
+        self._drop_cols_by_name(table)
+        self._remove_ignored_non_header_rows(table)
+        self._replace_dropdown_placeholder(table)
+        if self.drop_empty_rows:
+            self._drop_empty_rows(table)
+        if self.drop_empty_tables and table.df.empty:
+            table.df = None
+
+    def _lift_header(self, table: Table) -> None:
+        header = table.df.iloc[: self.num_header_rows]
+        header = self._concatenate_headers(header, headers_to_ffill=self.merged_header_rows)
+        table.df.columns = header
+        table.df = table.df.iloc[self.num_header_rows :]
+        table.df = table.df.reset_index(drop=True)
+        table.first_row_idx += self.num_header_rows
+
+    @staticmethod
+    def _remove_merged_headers(table: Table) -> None:
+        prev_col = ("", "")
+        table.col_idx_map = dict(
+            [
+                prev_col := (col_name, prev_col[1] if col_name == prev_col[0] else idx)  # noqa
+                for idx, col_name in enumerate(table.df.columns)
+            ]
+        )
+        duplicated = table.df.columns.duplicated()
+        table.df = table.df.loc[:, ~duplicated]
+
+    @staticmethod
+    def _concatenate_headers(header: pd.DataFrame, headers_to_ffill: list[int]) -> list:
+        """Fills null cells and concatenates columns of values to produce a single header per column.
+
+        Forward fill is necessary for some rows because merged cells in Excel files are read into pandas as individually
+        split cells with the left-most cell containing the original contents and subsequent merged cells being left as
+        null. Forward fill replaces these empty values with the original merged cell value.
+
+        :param header: pd.DataFrame of rows considered to contain header information
+        :param headers_to_ffill: indexes of header rows to ffill
+        :return: a list of concatenated headers
+        """
+        for row_idx in headers_to_ffill:
+            header.iloc[row_idx, :] = header.iloc[row_idx, :].fillna(method="ffill")
+        header = header.fillna("")
+        concatenated_headers = header.apply(lambda x: ", ".join([s for s in x if s]))
+        return concatenated_headers
+
+    def _drop_cols_by_name(self, table: Table) -> None:
+        if missing_cols := [col for col in self.col_names_to_drop if col not in table.df.columns]:
+            raise ValueError(f"Column(s) to drop missing from table - {missing_cols}")
+        table.df = table.df.drop(columns=self.col_names_to_drop, axis=1)
+        for col in self.col_names_to_drop:
+            del table.col_idx_map[col]
+
+    def _remove_ignored_non_header_rows(self, table: Table) -> None:
+        if any(idx not in table.df.index for idx in self.ignored_non_header_rows):
+            raise ValueError("Ignored non-header rows missing from table")
```

Comment:
This error message could be made more explicit by saying something that mentions indices and "out-of-bounds"